### PR TITLE
526 reserves info mg

### PR DESCRIPTION
**Pull request by markgreenburg**
- Adds reserve / national guard page to the 526 form.
- The page only shows if one or more service periods include 'National Guard' or 'Reserve' branches
- Title 10 activation date is conditionally expanded if the veteran indicates they are currently activated under title 10.
- Anticipated separation dates must be in the future
- Obligation dates (are different than service dates) are prefilled when available
- The page description pulls out the _latest_ reserves / ng period name & end date from the user's service history array

screen shot 2018-07-27 at 2 57 31 pm

----------

screen shot 2018-07-27 at 2 57 45 pm

----------

screen shot 2018-07-27 at 3 40 20 pm

### CODE REVIEW

**Review by peggygannon — CHANGES_REQUESTED** [review]
Thanks, @markgreenburg Please see my suggested edits below. Can you show me an updated screenshot after you input the changes? 

Please use this format for the date: 
July 20, 2014 (no ordinal numbers in dates)

Please see edits below for the labels and checkbox text: : 
Unit name
Obligation start date
Obligation end date

I'm currently activated on federal orders

I choose to waive VA compensation pay for the days I receive inactive duty training pay, so I can keep my inactive duty training pay.

Activation date
Expected separation date

error: Expected separation date must be in the future

**Comment by markgreenburg** [comment]
Thanks for the review @peggygannon - here's an updated screenshot:

screen shot 2018-07-30 at 10 42 50 am

**Review by peggygannon — APPROVED** [review]

File: src/applications/disability-benefits/526EZ/tests/helpers.unit.spec.jsx [hunk 1]
```diff
@@ -183,7 +229,7 @@ describe('526 helpers', () => {
     it('should return original data if disabilities is not an array', () => {
       const clonedData = _.cloneDeep(initialData);
       const pages = [];
-      const formData = _.set(clonedData, 'disabilities', { someProperty: 'value' });
+      const formData = _.set('disabilities', clonedData, { someProperty: 'value' });
```
Comment (cvalarida) [inline src/applications/disability-benefits/526EZ/tests/helpers.unit.spec.jsx]: This is using the normal lodash [`_.set()`](https://lodash.com/docs/4.17.10#set) (not `fp`), so I think we want the object first, then the path.

File: src/applications/disability-benefits/526EZ/tests/helpers.unit.spec.jsx [hunk 2]
```diff
@@ -316,4 +380,144 @@ describe('526 helpers', () => {
       });
     });
   });
+
+  describe('hasGuardOrReservePeriod', () => {
+    it('should return true when reserve or guard period present', () => {
```
Comment (cvalarida) [inline src/applications/disability-benefits/526EZ/tests/helpers.unit.spec.jsx]: Should we split these tests out? The way it is now, it only tests the reserve bit, not the guard. It's pretty duplicative, but maybe worthwhile?

**Review by cvalarida — APPROVED** [review]
I left a lot of comments, but they're all pretty minor. I'll go ahead and approve this so it doesn't hold you up, but there are a couple things that should probably be addressed before merging.

File: src/applications/disability-benefits/526EZ/helpers.jsx [hunk 3]
```diff
@@ -55,6 +57,50 @@ const setPhoneEmailPaths = (veteran) => {
...
+    waiveVABenefitsToRetainTrainingPay
+  } = formData;
+
+  // Ensure all required fields are present
+  if (
+    !unitName
+    || !obligationTermOfServiceDateRange
+    || !obligationTermOfServiceDateRange.from
+    || !obligationTermOfServiceDateRange.to
+    || typeof waiveVABenefitsToRetainTrainingPay === 'undefined'
```
[the remark this answers is not in the document]
Comment (markgreenburg) [reply src/applications/disability-benefits/526EZ/helpers.jsx]: sure is

File: src/applications/disability-benefits/526EZ/pages/reservesNationalGuardService.jsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import fullSchema526EZ from 'vets-json-schema/dist/21-526EZ-schema.json';
+// import fullSchema526EZ from '/local/path/vets-json-schema/dist/21-526EZ-schema.json';
```
[the remark this answers is not in the document]
Comment (markgreenburg) [reply src/applications/disability-benefits/526EZ/pages/reservesNationalGuardService.jsx]: Eh...unfortunately I'm still using it all the time 😞

File: src/applications/disability-benefits/526EZ/helpers.jsx [hunk 5]
```diff
@@ -834,3 +897,66 @@ export const PaymentDescription = () => (
...
+    const { nationalGuard, reserve } = RESERVE_GUARD_TYPES;
+    return (serviceBranch.includes(nationalGuard) || serviceBranch.includes(reserve));
+  }).map(({ serviceBranch, dateRange }) => {
+    const dateTo = new Date(dateRange.to);
+    return {
+      serviceBranch,
+      to: dateTo
+    };
+  }).sort((periodA, periodB) => {
+    if (periodA.to < periodB.to) { // could probably simplify sort to b.to - a.to
```
[the remark this answers is not in the document]
Comment (markgreenburg) [reply src/applications/disability-benefits/526EZ/helpers.jsx]: Don't know how I feel about you using my comments against me 😛

File: src/applications/disability-benefits/526EZ/tests/helpers.unit.spec.jsx (hunk 2 above)
Comment (markgreenburg) [reply src/applications/disability-benefits/526EZ/tests/helpers.unit.spec.jsx]: 🤔 I mean, I don't see the harm...might as well

File: src/applications/disability-benefits/526EZ/tests/helpers.unit.spec.jsx (hunk 1 above)
Comment (markgreenburg) [reply src/applications/disability-benefits/526EZ/tests/helpers.unit.spec.jsx]: 😬

File: src/applications/disability-benefits/526EZ/helpers.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (cvalarida) [reply src/applications/disability-benefits/526EZ/helpers.jsx]: :innocent:

**Pull request merged**